feat: documentation update (#809)
philips-software/amp-embedded-infra-lib: documents/modules/ROOT/pages/CodingStandard.adoc

== Introduction and Scope

> This coding standard defines various aspects of the contents of C++ source files. The goal is to provide a usable standard from which programmers should only deviate in very specific circumstances. Care shall be taken to not restrict programmers in any way or form, so that new techniques and new insights are not obstructed by this standard.

Reviewer: The intention here is to restrict programmers in form, and nothing else than form.

```suggestion
This coding standard defines various aspects of the contents of C++ source files. The goal is to provide a usable standard from which programmers should only deviate in very specific circumstances. Care shall be taken to not restrict programmers in any way but form, so that new techniques and new insights are not obstructed by this standard.
```

Author: Ok! Was not clear at first. Still, to make it clearer:

```suggestion
This coding standard defines various aspects of the contents of C++ source files. The goal is to provide a usable standard from which programmers should only deviate in very specific circumstances. Care shall be taken to not restrict programmers in any other way but form, so that new techniques and new insights are not obstructed by this standard.
```

Reviewer: How about "any other than form"?

Author: Sounds good!

----

[NOTE]

> `#ifdef`, `#if`, `#else`, `#elif` does not need to be followed by indentation, although there are certain cases where indentation would improve clarity.

Reviewer: ```suggestion
`#ifdef`, `#if`, `#else`, `#elif` do not need to be followed by indentation, although there are certain cases where indentation would improve clarity.
```

*Rule 17.* A logical block consists of lines of code that logically belong together and shall not contain blank lines. Logical blocks are separated from other blocks by one blank line.

*Rule 18.* A blank line shall not be followed by another blank line.

> *Rule 19.* A `{` belongs to the logical block beneath it, and a `}` belongs to the logical block above it, and shall therefore not be succeeded or preceded by a blank line, respectively.

Reviewer: ```suggestion
*Rule 19.* A `{` belongs to the logical block beneath it, and a `}` belongs to the logical block above it, and shall therefore not be succeeded nor preceded by a blank line.
```

    : SomeVeryLargeFactoryFunctionThatTranslatesColours(green);
----

> *Rule 42.* `{` and `}` shall be the single token in one line, except when used to define short arrays, when used in a lambda expressions containing only a single statement, and when used in an empty definition. Example:

Reviewer: Do you want to argue here that we should write a class like this?
class C
{
}
;

Author: No. Just thought it was redundant and/or confusing. Perhaps:

```suggestion
*Rule 42.* `{` and `}` shall be the single token in one line, except when `}` is followed by `;`, e.g. when defining short arrays, with lambda expressions containing a single statement, and when used in an empty definition. Example:
```

Reviewer: The intention is still a bit different. Most of the times, { and } appear on a single line. What follows are a number of exceptions to this rule. The first exception is than ; can be placed right after }. The other exceptions are when { and } are together on one line.

Author: I see. My confusion was that the other exceptions listed also comply with the '}' followed by ';' . Maybe using a single 'exception' statement:

```suggestion
*Rule 42.* `{` and `}` shall be the single token in one line, except when `}` is followed by `;`,  when used to define short arrays, when used in a lambda expressions containing only a single statement, and when used in an empty definition. Example:
```
